secrets/field.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2025 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"go.yaml.in/yaml/v2"
+)
+
+// Field is a field containing a secret.
+type Field struct {
+	providerName   string
+	providerConfig ProviderConfig
+	manager        *Manager
```

Comment:
idea during 1:1, it would be nice to remove bi-direction and have lazy refresh perhaps

Reply:
Alternatively just use interface for Refresh()